test/inductor/test_flex_attention.py

```diff
@@ -108,6 +109,11 @@ def _trig2(score, b, h, m, n):
     return z
 
 
+# --------- Useful mask functions for testing ---------
+def _causal_mask(batch, head, token_q, token_kv):
```

Comment:
I wonder if we should just make this have the same function signature as `score_mod`. It's a little bit awkward, but arguably having a single function signature makes it simpler.

Reply:
Yea, I think if we make ```score_mod``` and ```mask``` have the same signature, it would be simpler.

Reply:
Discussed offline: ```score_mod``` and ```mask_fn``` should be different signatures. For ```create_block_mask``` function, advanced users would provide ```mask_fn``` for better performance, and fall back to non partial mask routine if users don't provide ```mask_fn```.